src/purify.spec.ts

```diff
@@ -1,16 +1,104 @@
-import { oneLine } from 'common-tags';
+import { oneLine, stripIndent } from 'common-tags';
 
 import { purify } from './purify';
 
 
-// TODO: add purify tests
 describe('purify', () => {
-  it('TODO', () => {
-    const input = oneLine`
+  it('wraps ts 2.2 enums in IIFE', () => {
+    const input = stripIndent`
+      var ChangeDetectionStrategy = {};
+      ChangeDetectionStrategy.OnPush = 0;
+      ChangeDetectionStrategy.Default = 1;
+      ChangeDetectionStrategy[ChangeDetectionStrategy.OnPush] = "OnPush";
+      ChangeDetectionStrategy[ChangeDetectionStrategy.Default] = "Default";
     `;
-    const output = oneLine`
+    const output = stripIndent`
+      var ChangeDetectionStrategy = /*@__PURE__*/(function() {
+      var ChangeDetectionStrategy = {};
+      ChangeDetectionStrategy.OnPush = 0;
+      ChangeDetectionStrategy.Default = 1;
+      ChangeDetectionStrategy[ChangeDetectionStrategy.OnPush] = "OnPush";
+      ChangeDetectionStrategy[ChangeDetectionStrategy.Default] = "Default";;
+      return ChangeDetectionStrategy;})();
     `;
 
-    expect(oneLine`${purify(input)}`).toEqual(output);
+    expect(oneLine`${purify(input)}`).toEqual(oneLine`${output}`);
+  });
+
+  it('wraps ts 2.3 enums in IIFE', () => {
+    const input = stripIndent`
+      var ChangeDetectionStrategy;
+      (function (ChangeDetectionStrategy) {
+        ChangeDetectionStrategy[ChangeDetectionStrategy["OnPush"] = 0] = "OnPush";
+        ChangeDetectionStrategy[ChangeDetectionStrategy["Default"] = 1] = "Default";
+      })(ChangeDetectionStrategy || (ChangeDetectionStrategy = {}));
+    `;
+    const output = stripIndent`
+      var ChangeDetectionStrategy = /*@__PURE__*/(function() {
+          var ChangeDetectionStrategy = {};
+          ChangeDetectionStrategy[ChangeDetectionStrategy["OnPush"] = 0] = "OnPush";
+          ChangeDetectionStrategy[ChangeDetectionStrategy["Default"] = 1] = "Default";
+          return ChangeDetectionStrategy;
+      })();
+    `;
+
+    expect(oneLine`${purify(input)}`).toEqual(oneLine`${output}`);
+  });
+
+  it('prefixes safe imports with /*@__PURE__*/', () => {
+    const input = stripIndent`
+      /* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__angular_http__ = __webpack_require__(72);
```

Comment:
Add a test for if you're using the `HashedModuleIdsPlugin`? e.g.  `webpack_require("EE82");`